libs/utils/energy.py

```diff
@@ -275,22 +242,32 @@ def reset(self):
         self._log.debug('RESET: %s', self.channels)
 
         self._aep.start()
-        self.time['start'] = time.time()
 
     def report(self, out_dir, out_energy='energy.json', out_samples='samples.csv'):
-        if self._aep is None:
-            return
+        self._aep.stop()
 
-        # Retrieve energy consumption data
-        csv_file = '{}/{}'.format(out_dir, out_samples)
-        self._sample(csv_file)
+        csv_path = os.path.join(out_dir, out_samples)
+        self._aep.get_data(csv_path)
+        with open(csv_path) as f:
+            # Each column in the CSV will be headed with 'SITE_measure'
+            # (e.g. 'BAT_power'). Convert that to a list of ('SITE', 'mesure)
+            # tuples, then pass that as the `names` parameter to read_csv to get
+            # a nested column index. None of devlib's standard measurement types
+            # have '_' in the name so this use of rsplit should be fine.
+            headers = f.readline().strip().split(',')
+            columns = [tuple(h.rsplit('_', 1)) for h in headers]
+            # Passing `names` means read_csv doesn't expect to find headers in
+            # the CSV (i.e. expects every line to hold data). This works because
+            # we have already consumed the first line of `f`.
+            df = pd.read_csv(f, names=columns)
+
+        if df.empty:
+            raise RuntimeError('No energy data collected')
 
-        # Reformat data for output generation
         channels_nrg = {}
-        for channel in self.channels:
-            self._log.debug('Energy [%16s]: %.6f',
-                            channel.site, channel.nrg)
-            channels_nrg[channel.site] = channel.nrg
+        for site, measure in df:
+            if measure == 'power':
+                channels_nrg[site] = df[site]['power'].sum()
```

Comment:
Yeah this assumes a constant sampling rate, now I think about it I guess I've assumed that the meter automatically scales power according to the sample rate which is probably not true. 

As you said this doesn't matter for our purposes but let's try to get it right anyway. Devlib doesn't actually expose the sample rate right now so I'll try to add it.